lib/spells/effects/Damage.cpp

```diff
@@ -152,6 +152,22 @@ void Damage::describeEffect(std::vector<MetaString> & log, const Mechanics * m,
 		m->caster->getCasterName(line);
 		log.push_back(line);
 	}
+	else if(m->getSpell()->getJsonKey().find("accurateShot") != std::string::npos && !multiple)
+	{
+		MetaString line;
+		if(kills > 1)
+		{
+			line.appendTextID("vcmi.battleWindow.accurateShot.resultDescription"); //(number) (unit type) was killed with an accurate shot!
+			line.replaceNumber(kills);
+			firstTarget->addNameReplacement(line, true);
+		}
+		else
+		{
+			line.appendTextID("vcmi.battleWindow.accurateShot.resultDescription.1"); //1 (unit type) were killed by accurate shots!
+			firstTarget->addNameReplacement(line, false);
+		}
```

Comment:
- Change code to this
- Add translations for these strings:
```
vcmi.battleWindow.accurateShot.resultDescription.0 
vcmi.battleWindow.accurateShot.resultDescription.1
vcmi.battleWindow.accurateShot.resultDescription.2
```
And game will pick correct form even for languages like Polish
(won't work correctly if client and server have different languages at the moment, will look into it later)

```suggestion
		std::string preferredLanguage = VLC->generaltexth->getPreferredLanguage();
		std::string textID = "vcmi.battleWindow.accurateShot.resultDescription";
		line.appendTextID(Languages::getPluralFormTextID( preferredLanguage, kills, text));
		line.replaceNumber(kills);
		firstTarget->addNameReplacement(line, true);
```